### PR TITLE
Build improvements & Travis CI integration

### PR DESCRIPTION
**Pull request by niksajakovljevic**
Build:
- add build target and always build a new artifact
- check if `golang/dep` is present and fetch dependencies
- run tests on all packages

Add basic Travis CI for build and test.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,10 @@
+language: go
+
+go:
+  - 1.10.x
+
+before_install:                                                                 
+  - curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh
```
Comment (erimatnor) [inline .travis.yml]: Is this step really needed given the `godep` support in Travis: https://docs.travis-ci.com/user/languages/go/#godep-support?

If it's needed, add a comment explaining what this does. (I initially thought that this installed Golang).

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,10 @@
+language: go
+
+go:
+  - 1.10.x
+
+before_install:                                                                 
+  - curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh  
+  - dep ensure
+
+script: make build test
```
Comment (erimatnor) [inline .travis.yml]: Add new line.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,10 @@
+language: go
+
+go:
+  - 1.10.x
+
+before_install:                                                                 
+  - curl https://raw.githubusercontent.com/golang/dep/master/install.sh | sh  
+  - dep ensure
```
Comment (erimatnor) [inline .travis.yml]: Is this needed? Isn't `dep ensure` run in via the `Makefile`?

File: Makefile [hunk 4]
```diff
@@ -26,8 +23,12 @@ version.properties:
 .target_os:
 	@echo $(OS) > .target_os
 
-$(TARGET): .target_os $(SOURCES)
-	GOOS=$(OS) GOARCH=${ARCH} CGO_ENABLED=0 go build -a -installsuffix cgo --ldflags '-w' -o $@ main.go
+build: $(TARGET)
+
+$(TARGET): .target_os
```
Comment (erimatnor) [inline Makefile]: If you remove `$(SOURCES)`  from the target dependencies, will it rebuild if you modify a source file?

**Review by erimatnor — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 1 above)
Comment (niksajakovljevic) [reply .travis.yml]: Travis does not support `golang\dep` out of the box ( I assume they should fix it as more and more people are migrating to `golang\dep`). I'll add a comment though

File: .travis.yml (hunk 3 above)
Comment (niksajakovljevic) [reply .travis.yml]: Currently `dep ensure` runs in `make build`, so we need it for `make test` to work.

File: Makefile (hunk 4 above)
Comment (niksajakovljevic) [reply Makefile]: Since I added `$(TARGET)` to `.PHONY` it will always trigger

**Pull request merged**